lib/idempo.rb

```diff
@@ -58,6 +59,12 @@ def call(env)
 
       expires_in_seconds = (headers.delete("X-Idempo-Persist-For-Seconds") || @persist_for_seconds).to_i
       if response_may_be_persisted?(status, headers, body)
+        if Gem::Version.new(Rack.release) >= Gem::Version.new("3.0")
+          # `body` could be of type `ActionDispatch::Response::RackBody` and idempo will not even attempt to store it,
+          # we're converting ActionDispatch::Response::RackBody to a storable array format.
+          body = body.try(:to_ary) || body if !body.is_a?(Array) && !body.is_a?(Enumerator)
```

Comment:
No need for any extra checkes except for the responding to `to_ary`. An Array will respond to `to_ary` and return `self`:

```ruby
[2] pry(main)> ar = []
=> []
[3] pry(main)> ar.object_id
=> 320
[4] pry(main)> ar.to_ary.object_id
=> 320
```

A Body being an Enumerator is another check that should not be done, as an object can perfectly be a Rack response body and respond to either `each` or `call`, but not _be_ an Enumerator.